### PR TITLE
Archive datasource

### PR DESCRIPTION
**Pull request by shroffk**
This is a WIP in preparation of the aa datasource project for the upcoming codeathon

### CODE REVIEW

File: app/trends/archive-datasource/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
+Alarm Datasource
```
Comment (kasemir) [inline app/trends/archive-datasource/doc/index.rst:1]: Archive, not Alarm Datasource

**Comment by shroffk** [comment]
This PR includes a restructuring of the databrowser application.

I wanted to move the archive reader bits out of the databrowser app since there are a few use cases where we see the archive reader being used without needing the UI bits. The foremost of these use cases is the archive datasource which is going to be used in the Save and Restore service and potentially also via the PVWS.

I think there might be more ideas to consider on how to restructure this...should each archiver have its reader in a standalone module, how much of the utility stuff should be move to core modules ( I already moved some of the Timestamp support classes to core-util), etc... I will add this PR as a discussion topic for next weeks codeathon

File: app/trends/archive-datasource/doc/index.rst (hunk 1 above)
Comment (shroffk) [reply app/trends/archive-datasource/doc/index.rst]: fixed this.. thanks

**Review by georgweiss — COMMENTED** [review]
The ```Preferences``` class needs to change:
```AnnotatedPreferences.initialize(Activator.class, Preferences.class, "/appliance_datasource_preferences.properties");```
to
```AnnotatedPreferences.initialize(Preferences.class, "/appliance_datasource_preferences.properties");```
otherwise setting in custom ini file will be ignored.

Looks like preference ```archive_url``` supports but one URL, unlike preference for data browser. At ESS we currently use four URLs.

Maybe the doc file should state supported time formats, and also clarify that the time specifier should be unquoted.

Seems time span is not (yet) supported?

**Comment by shroffk** [comment]
I had fixed the preference but forgot to commit.

I believe that the documentation needs a lot more work... I hoping this to be accomplished during the codeathon

If we want to support multiple then we will have to have a system of selecting/setting one by default and then having some parameter to select from the rest... or should we let the user define any URL?

**Comment by shroffk** [comment]
Currently, 
I restructured this so that all the archive readers are still in one common module. 
@kasemir @georgweiss does it make sense to split this too? should we have one dedicated module for each of the archive readers?

**Comment by kasemir** [comment]
At this time I see no advantage in splitting it further. Just adds more build steps.

**Comment by shroffk** [comment]
Ok,
So we keep this PR as it is.
In the future if we want to clean up the dependencies in our individual products we can revisit this.

**Review by georgweiss — COMMENTED** [review]
Looks good, really nice feature. 
Replay can easily be tested in probe. However, replay seems to push updates as "quickly" as possible. Should not time stamps be used to replay at the pace events were stored in the archiver?

**Comment by kasemir** [comment]
> Should not time stamps be used to replay at the pace events were stored in the archiver?

Great idea! Should also be applied to the data browser:
If user selects the last year as the time range, the plot will populate sample by sample over the course of one year. That way, you can most realistically re-live what happened over time ;-)

**Comment by georgweiss** [comment]
> the plot will populate sample by sample over the course of one year.
In the meanwhile we can spend time in developers' meetings.

**Comment by shroffk** [comment]
Well... the idea would be to have a rate/multiplier so basically it would be replay that hour I am interested in at 50x or 10x etc...

**Comment by kasemir** [comment]
Actually, you have to slow it down further. The original changes were clearly too fast, that's why users missed them and now need to re-inspect. By playing them back at 1/10 or 1/100 of the original speed, users will be able to catch up with what happened.

**Review by georgweiss — APPROVED** [review]

**Comment by shroffk** [comment]
Sorry I accidentally merged this, I had planned to do this next week

@kasemir @katysaintin @tynanford 
Your products will have to be updated to include the archive reader module which has the implementations of the archive reader SPI's

**Pull request merged**